mfr/server/handlers/core.py

```diff
@@ -132,6 +132,7 @@ def NAME(self):
         )
 
         self.source_file_id = uuid.uuid4()
+        self.add_header('X-MFR-REQUEST-ID', str(uuid.uuid4()))
```

Comment:
I have a few further questions on how this work.

1. Do servers (WB and MFR) need to handle this customized header explicitly? For example, is it automatically logged?

2. Why not let client generate the header? (WB and MFR can generate it only when client does not). We cannot guarantee that the client always receives a response that contains this header, especially when exceptions occur.

3. For "**It would be doubly nice if somehow we could propagate this request id when a request involves multiple services.**": can we pass this ID along the requests pipeline (including requests to different OSF services which belong to the same request cycle?

4. `X-REQUEST-ID` is a the general name. What's the pros and cons of using `X-MFR-REQUEST-ID` and `X-WB-REQUEST-ID`?

Reply:
My understanding is that the immediate use for this is that it is visible in the sentry logs, so if a user can give us their request id we can easily associate that with the logs, other then that other uses are speculative.